packages/server/lib/cloud/api.ts

```diff
@@ -353,5 +453,30 @@ module.exports = {
     responseCache = {}
   },
 
+  preflight (preflightInfo) {
+    return retryWithBackoff(async (attemptIndex) => {
+      const preflightBase = process.env.CYPRESS_API_URL ? apiUrl.replace('api', 'api-proxy') : apiUrl
+      const result = await rp.post({
+        url: `${preflightBase}preflight`,
+        body: {
+          apiUrl,
+          envUrl: process.env.CYPRESS_API_URL,
+          ...preflightInfo,
+        },
+        headers: {
+          'x-route-version': '1',
+          'x-cypress-request-attempt': attemptIndex,
+        },
+        json: true,
+        encrypt: 'always',
+      })
+
+      preflightResult = result // { encrypt: boolean, apiUrl: string }
+      recordRoutes = makeRoutes(result.apiUrl)
```

Comment:
Why would the route be changing vs using our record routes with some sort of auth token on requests?